scripts/kubestellar-snapshot.sh

```diff
@@ -0,0 +1,567 @@
+        fi
+    fi
+    cp_n=$((cp_n+1))
+done
+
+
+###############################################################################
+# Listing managed clusters
+###############################################################################
+echotitle "Managed Clusters:"
+mc_n=0
+for j in "${!cp_pch[@]}" ; do
+    if [[ "${cp_pch[$j]}" == "its" ]] ; then
+        mcs=($(kubectl --kubeconfig <(echo "${cp_kubeconfig[$j]}") get managedcluster -no-headers -o name 2> /dev/null || true))
+        for i in "${!mcs[@]}" ; do
+            name=${mcs[i]##*/}
+            mc_name[mc_n]=$name
+            accepted="$(kubectl --kubeconfig <(echo "${cp_kubeconfig[$j]}") get managedcluster $name -o jsonpath='{.status.conditions}' | jq '.[] | select(.type == "HubAcceptedManagedCluster") | .status' | tr -d '"')"
+            joined="$(kubectl --kubeconfig <(echo "${cp_kubeconfig[$j]}") get managedcluster $name -o jsonpath='{.status.conditions}' | jq '.[] | select(.type == "ManagedClusterJoined") | .status' | tr -d '"')"
+            available="$(kubectl --kubeconfig <(echo "${cp_kubeconfig[$j]}") get managedcluster $name -o jsonpath='{.status.conditions}' | jq '.[] | select(.type == "ManagedClusterConditionAvailable") | .status' | tr -d '"')"
+            synced="$(kubectl --kubeconfig <(echo "${cp_kubeconfig[$j]}") get managedcluster $name -o jsonpath='{.status.conditions}' | jq '.[] | select(.type == "ManagedClusterConditionClockSynced") | .status' | tr -d '"')"
+            echo -e "- ${COLOR_INFO}${mc_name[mc_n]}${COLOR_NONE} in ${COLOR_INFO}${cp_name[j]}${COLOR_NONE}: accepted=$(echostatus $accepted), joined=$(echostatus $joined), available=$(echostatus $available), synced=$(echostatus $synced)"
+            if [[ "$arg_verbose" == "true" ]] ; then
+                echo -n -e "${COLOR_YAML}"
+                kubectl --kubeconfig <(echo "${cp_kubeconfig[$j]}") get managedclusters $name -o jsonpath='{.metadata.labels}' | jq '. |= with_entries(select(.key|(contains("open-cluster-management")|not)))' | indent
+                echo -n -e "${COLOR_NONE}"
+            fi
+            if [[ "$arg_yaml" == "true" ]] ; then
+                mkdir -p "$output_folder/${cp_name[$j]}/managed-clusters"
+                kubectl --kubeconfig <(echo "${cp_kubeconfig[$j]}") get managedcluster $name -o yaml > "$output_folder/${cp_name[$j]}/managed-clusters/$name.yaml"
+            fi
+            mc_n=$((mc_n+1))
+        done
+    fi
+done
+
+
+###############################################################################
+# Listing binding policies
+###############################################################################
+echotitle "Binding Policies:"
+bp_n=0
+for j in "${!cp_pch[@]}" ; do
+    if [[ "${cp_pch[$j]}" == "wds" ]] ; then
+        bps=($(kubectl --kubeconfig <(echo "${cp_kubeconfig[$j]}") get bindingpolicy -no-headers -o name 2> /dev/null || true))
```

Comment:
Why is an array being built? I only see use of the current (i.e., currently last) element.

Reply:
It was done this way to match the other sections.
It's true that it's not currently being used but I would rather keep it for possible future if it's not a big deal

Reply:
No, it is not a big deal.